src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -427,6 +430,94 @@ public override bool ValidateChildren
             }
         }
 
+        /// <inheritdoc />
+        public override bool IsValidationRequired
+        {
+            get
+            {
+                if (!_isValidationRequired.HasValue)
+                {
+                    CalculateIsValidationRequired();
+                }
+
+                Debug.Assert(_isValidationRequired.HasValue);
+                return _isValidationRequired.Value;
+            }
+        }
+
+        private void CalculateIsValidationRequired()
```

Comment:
This method should return  bool rather than setting it directly. Using a field as a 'local' is a good way to end up with a threading bug.